Detectors/ITSMFT/common/reconstruction/include/ITSMFTReconstruction/RawPixelReader.h

```diff
@@ -192,6 +192,38 @@ class RawPixelReader : public PixelReader
     return getNextChipData(chipDataVec); // is it ok to use recursion here?
   }
 
+  ChipPixelData* getNextChipDataFromBuffer(std::vector<ChipPixelData>& chipDataVec)
+  {
+    // decode new RU if no cached non-empty chips
+
+    if (mCurRUDecodeID >= 0) { // make sure current RU has fired chips to extract
+      for (; mCurRUDecodeID < mNRUs; mCurRUDecodeID++) {
+        auto& ru = mRUDecodeVec[mCurRUDecodeID];
+        if (ru.lastChipChecked < ru.nChipsFired) {
+          auto& chipData = ru.chipsData[ru.lastChipChecked++];
+          int id = chipData.getChipID();
+          chipDataVec[id].swap(chipData);
+          return &chipDataVec[id];
+        }
+      }
+      mCurRUDecodeID = 0; // no more decoded data if reached this place,
+    }
+    // will need to decode new trigger
+    if (!mDecodeNextAuto) { // no more data in the current ROF and no automatic decoding of next one was requested
+      return nullptr;
+    }
+
+    if (!mRawBuffer.isEmpty()) {
```

Comment:
this method fully duplicates the getNextChipData method except this line instead of the check ``if (mMinTriggersCached < 2)`` and uses clone of cacheLinksData with actual data loading commented.
What is the reason for that?
Collecting the statistics for the trigger could be done in existing routines.

Reply:
Hi @shahor02, thanks for the quick feedback. I leave @zla2019 to answer your questions and comments as he wrote these parts. Thank you.

Reply:
Yes, because the getNextChipData method is reading raw data from a file. And what we need is reading raw data from a buffer. So I want to change some code to do this.

Reply:
But you are using PayLoadCont which has to be somehow filled. I would strongly advize to switch to the ``o2-itsmft-stf-decoder-workflow``  which can decode directly STF messages, w/o any intermediate files.

Reply:
Yes, I am setting PayLoadCont address before use the getNextChipData method. I will try to use o2-itsmft-stf-decoder-workflow.